### ISSUE TITLE
`manage_relationship(type: :replace)` seems to be both deprecated and removed in 2.0

### ISSUE THREAD

**Issue by sevenseacat** [ash-project/ash#381]
**Describe the bug**

Couple of issues:

* Upgrade docs don't mention `type: replace` being deprecated (only that `replace_relationship/4` is deprecated (which I didn't even know existed)
* After updating to latest revision in Ash main and starting my app, I get the following:

```
13:47:30.732 [warning] type: :replace has been renamed to `:append_and_remove` in 2.0, and it will be removed in 2.1

13:47:30.793 [warning] type: :replace has been renamed to `:append_and_remove` in 2.0, and it will be removed in 2.1

== Compilation error in file lib/myapp/ash/resources/element/confirmation.ex ==
** (Spark.Error.DslError) [MyApp.Element.Confirmation]
 actions -> update -> update -> change -> manage_relationship -> media_item_id -> media_item:
  The following error was raised when validating options provided to manage_relationship.

** (NimbleOptions.ValidationError) expected :type to be in [:append_and_remove, :append, :remove, :direct_control, :create], got: :replace
    (nimble_options 0.4.0) lib/nimble_options.ex:291: NimbleOptions.validate!/2
    (ash 2.0.0-rc.9) lib/ash/resource/transformers/validate_manage_relationship_opts.ex:64: anonymous fn/3 in Ash.Resource.Transformers.ValidateManagedRelationshipOpts.transform/1
    ...
```

The warning implies its fine I can still use `type: :replace` for now, but the compilation error says 🙅 I may not use it.

**To Reproduce**

Define an action that uses `manage_relationship(type: :replace)` - mine is something like:

```
    create :create do
      primary? true

      argument :element_id, :uuid
      argument :media_item_id, :uuid

      argument :options, :string

      change manage_relationship(:element_id, :element, type: :replace)
      change manage_relationship(:media_item_id, :media_item, type: :replace)
    end
```

**Expected behavior**

I'd expect either a warning or error - not both!

**Runtime**
 - Elixir version 1.14.0
 - Erlang version 25.0.4
 - OS MacOS Monterey 12.4
 - Ash version `main` branch, SHA `a802bf17fa0f6c030c382b316b950fe3b15cc343`
 - any related extension versions - `main` branch for all of ash, ash_admin, ash_phoenix, ash_postgres

**Comment by zachdaniel** [comment]
Ah, thanks for the report, easy fix, will take care of it tomorrow. For now the workaround of just replacing `type: :replace` with `type: :append_and_remove` is an option.

**Comment by zachdaniel** [comment]
Thanks for the report! fixed in 739bf5230989a6b04895bafb3deb478da5677246
